include/swift/SIL/GraphFunctionDeviceInfo.h

```diff
@@ -29,36 +31,61 @@ namespace tf {
 class GraphOperationBuilder;
 struct GraphOperationInfo;
 
-/// The device of a tfop instruction (and its output tensors, if any).
+/// The device type of a tfop instruction (and its output tensors, if any).
 enum class DeviceType {
   INVALID,
   CPU,
   GPU,
   TPU,
   /// Indicates this instruction should run on all devices in
-  /// `GraphFunctionDeviceInfo::usedDeviceTypes`. For example, a promoted
+  /// `GraphFunctionDeviceInfo::usedDeviceIds`. For example, a promoted
   /// scalar will run on all such devices, in case it is a loop iteration count
   /// and the loop runs on all devices.
   ALL,
 };
 
-static const char TF_DEFAULT_CPU_DEVICE[] =
-    "/job:localhost/replica:0/task:0/device:CPU:0";
-static const char TF_DEFAULT_GPU_DEVICE[] =
-    "/job:localhost/replica:0/task:0/device:GPU:0";
+/// The device id represents a local device, with two components: device type
+/// and index. An example is "GPU:3".
+// TODO: consider adding a worker (task) component to support remote devices.
+struct DeviceId {
```

Comment:
Nit: I think the local code base prefers `ID` to `Id`, e.g. `Lexer::getBufferID`.

Reply:
Agree that `ID` would also work. In this case since I've plumbed through using `DeviceId` in many call-sites and files, I'll leave it as is. Thanks for the suggestion though!